pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -73,6 +76,44 @@ func (n *nsLifecycle) syncNS(obj *v1.Namespace) (bool, error) {
 	return hasPRTBs, nil
 }
 
+func (n *nsLifecycle) assignToSystemProject(ns *v1.Namespace) error {
+	defaultProjectsToNamespaces, err := GetDefaultProjectsToNamespaces()
+	if err != nil {
+		return err
+	}
+	for projectDisplayName, namespaces := range defaultProjectsToNamespaces {
+		for _, nsToCheck := range namespaces {
+			if nsToCheck == ns.Name {
+				projectID := ns.Annotations[projectIDAnnotation]
+				if projectID != "" {
+					return nil
+				}
+
+				projects, err := n.m.projectLister.List(n.m.clusterName, labels.NewSelector())
```

Comment:
AH! This will get all projects across ALL clusters. You're going to do some really funky cross-cluster assigning.

It also seems very inefficient to loop over ALL projects to find the default one. Could you added a label so you can do a label selector or could you use an indexer?